### ISSUE TITLE
The communication part is abnormal

### ISSUE THREAD

**Issue by lixiaolx** [parasailteam/coconet#7]
When I use MNK(64,9216,4608) and run sample/matul-reduce.cu, I found a separate test and found the following problems. After exchanging the value of MN, I found that the final result after communication is wrong, and the communication time gap is large. Therefore, I suspect the communication kernel Is there a bug in it? Can you help me to look it up and explain it? The result is as follows
The result should be that running MNK(9216,64, 4608)  succeeded, but running MNK(64,9216, 4608) failed

```
NCCL INFO Launch mode Parallel
rankk -1 Mismatch at 0 : ref '36864.000000', computed '4608.000000'
rankk -1 Mismatch at 0 : ref '36864.000000', computed '4608.000000'
rankk -1 Mismatch at 0 : ref '36864.000000', computed '4608.000000'
rankk -1 Mismatch at 0 : ref '36864.000000', computed '4608.000000'
rankk -1 Mismatch at 0 : ref '36864.000000', computed '4608.000000'
rankk -1 Mismatch at 0 : ref '36864.000000', computed '4608.000000'
rankk -1 Mismatch at 0 : ref '36864.000000', computed '4608.000000'
checking results at iter 0 
rankk -1 Mismatch at 0 : ref '36864.000000', computed '4608.000000'
iter 20
iter 40
iter 60
iter 80
iter 100
AllReduce+cuBLASMatMul: channel 12 buffer_size 9437184 MUL [64, 4608] X [4608, 9216]: TotalTime 19.334793 ms, AllReduceTime 9.166956 ms, cuBLAS Time 8.098815 ms
```
```
[0] NCCL INFO Launch mode Parallel
checking results at iter 0 
iter 20
iter 40
iter 60
iter 80
iter 100
AllReduce+cuBLASMatMul: channel 12 buffer_size 9437184 MUL [9216, 4608] X [4608, 64]: TotalTime 28.081179 ms, AllReduceTime 17.275572 ms, cuBLAS Time 8.818693 ms

```

**Comment by abhijangda** [comment]
Hi, 

I am not surprised by it. AllReduce was modified heavily to do 2-D communication tiles and was mainly tested on GPT-2 and GPT-3 sizes. Supporting these sizes will require significant work. We are already working on 2-D tile communication in a separate project and I will share it once it is done.

**Comment by lixiaolx** [comment]
Okay, thank you very much. I'm looking forward to an optimized version of NCCL that can support different model sizes.

I am very interested in your project and would like to participate. Can I also ask you a few questions?
first.



> Supporting these sizes will require significant work. 

-  May I ask what work you have done with NCCL and have you modified the list of file names in the standard NCCL？

- If I want to test the effect on non GPt2/3, which files do I need to modify in https://github.com/parasailteam/coconet-experiments ?

- In your latest paper, you used cutlass in the calculation section, but the implementation of cutpass looks very complex. Can you please provide the nccl modification implementation corresponding to using cubals+ALLReduce (non cutpass in Gemm calculation section) in your first version?

**Comment by abhijangda** [comment]
I think you might be able to run it. Try following things for 64,9216, 4608:
1. There is "chunkCols" set in the example modify that to factors of 9216, such as 9*16, 9*32, .. 
2. Then try with different values of NCCL_BUFFSIZE,  NCCL_MAX_NCHANNELS and NCCL_MIN_NCHANNELS, but make sure that GPUS * NCCL_BUFFSIZE * NCCL_MAX_NCHANNELS < 64 * 9216. I suggest following values:
 
NCCL_BUFFSIZE =  9*256, 9*512, 9*1024, 9*2048, ...
NCCL_MAX_NCHANNELS = NCCL_MIN_NCHANNELS = 9, 18, 27 

For your other questions can you make a separate issue for them?
